### PR TITLE
Allow non-string values in keys

### PR DESCRIPTION
**Pull request by Gaardsholt**

### CODE REVIEW

File: vault/readSecret_test.go [hunk 1]
```diff
@@ -24,3 +24,55 @@ func TestReadSecretWrongPath(t *testing.T) {
...
+func TestReadSecretKeyWithNumberAsValue(t *testing.T) {
+	// arrange
+	path := "secret/data/secret"
+	key := "key4"
+	expectedValue := float64(123)
+
+	testReadSecretKey(path, key, expectedValue, t)
+}
+
+// TestReadSecretKeyWithBooleanAsValue tests that the function returns the correct secret key
```
Comment (Andrei-Predoiu) [inline vault/readSecret_test.go:52]: These comments might as well not be there 😄

File: vault/readSecret.go [hunk 2]
```diff
@@ -70,5 +70,6 @@ func (client *API) ReadSecretKey(path string, key string) (string, error) {
 		return "", fmt.Errorf(keyNotFound, key, path, err)
 	}
 
-	return secretKey.(string), nil
+	// return secretKey.(string), nil
```
Comment (Andrei-Predoiu) [inline vault/readSecret.go:73]:
```suggestion
```

**Review by Andrei-Predoiu — COMMENTED** [review]
What happens if the values is a json struct or something?

**Review by OrKarstoft — APPROVED** [review]
Only what Andrei mentioned.

File: vault/readSecret_test.go (hunk 1 above)
Comment (Gaardsholt) [reply vault/readSecret_test.go:52]: Yeah forgot to update them :D
Should be fixed now

**Comment by Gaardsholt** [comment]
> What happens if the values is a json struct or something?

I don't think, that has ever been supported tbh.
Do we have an example of anywhere where we are doing that?

**Review by Andrei-Predoiu — APPROVED** [review]
Not saying it was supported or anything, i'm just curious :)

**Comment by Gaardsholt** [comment]
But I think you are right, this will not work if the value is a JSON object.

**Pull request merged**